app/assets/javascripts/components/DecoratedInput.tsx

```diff
@@ -1,7 +1,8 @@
 import { FunctionalComponent, ComponentChild } from 'preact';
+import { HtmlInputTypes } from '@/enums';
 
 interface Props {
-  type?: string;
+  type?: HtmlInputTypes;
```

Comment:
This way, the user of the component will need to also import HtmlInputTypes instead of just passing a string that matches the string literal.

Reply:
Yeah, he will import the enum and use its values instead of typing that string.